Windows cmake msbuild

Jammyjamjamman opened the pull request:
Add new windows build method, using cmake + msbuild.

Jammyjamjamman committed:
Add script to install build deps using vcpkg.

Jammyjamjamman committed:
Cmake: Add build configuration for VS builds
* Build vs 2019 project using cmake.
* Cmake configured to work in windows. Hopefully still works in linux
too!
* Update vcpkg install script.
* Configures build for MG game release or debug mode.
* Still need to fix building mapeditor and g3d viewer.
* Remove CurlGetConfig.cmake (not needed!)

Jammyjamjamman committed:
Fix for linux and mac

Jammyjamjamman committed:
More progress with cmake for Visual Studio projects

* Change conditon so that L137 onwards of main CMakelists.txt is
executed fro VS project target. Main reason to do this was to make sure
project actually built with streflop.
* Change conditions after L137 so unwanted compile flags aren't added.
* Project now builds multi threaded for VS project. Tries to use max
number of cores.
* Simplify cxx flags. Remove unneeded MT and MTd flags.
* Set VS build output to mk/windoze.
* Add FPUSettings.asm to streflop in cmake, if VS project specified.
* Remove STREFLOP compile definitions. Not needed now streflop is
configured properly.
* Simplify cmake by adding "VS_BUILD" variable, instead of doing string
comparisons all the time to check if is "Visual Studio" target.

Jammyjamjamman committed:
Get map editor and model viewer building with VS cmake builds

* Map editor and model viewer build, but only as "Release" build.
* Unwanted flags not set for msbuild, such as FPIC, -g3, -O3, -sse...
* Wanted flags set for msbuild e.g. /arch:SSE2 /fp:fast -D__SSE__
-D__SSE2__
* Allow including wx, used for vcpkg wx.
* Mapeditor is WIN32 exe for VS_BUILD. (Not sure why this is needed...)
* g3d viewer and map editor mains edited to remove unnecessary windows-
only code(, for strings).

Jammyjamjamman committed:
New windows build helper scripts

* build-mg-vs-cmake.ps1 to build mg.
* clean-all.ps1 to clean mg.
* install-deps-vcpkg.ps1
* .gitignore ignores build and vcpkg folders.

Jammyjamjamman committed:
Fix build script errors

No longer errors when vcpkg path not specified.

Jammyjamjamman committed:
Fix map editor and g3d viewer build for linux and macos

Jammyjamjamman committed:
Windows build script: Remove unnecessary multithread build flag.

Jammyjamjamman committed:
More fixes to cmake builds

- Add icons to windows exes.
- Revert removal of CurlGetConfig.cmake. This is still required in
ubuntu. (Perhaps only for static builds.)

Jammyjamjamman committed:
Fix cmake-clang for macos

* Also fix for ubuntu 18, by decreasing required cmake version.

Jammyjamjamman committed:
Fix for windows + older linux distros

andy5995 reviewed (commented):
@Jammyjamjamman Do you need any help with the SDL includes?

Jammyjamjamman committed:
Improve build script

Jammyjamjamman committed:
some more improvements

Jammyjamjamman committed:
Fix windows g3d xml tools build

Jammyjamjamman committed:
Fix SSE check and cleanup

Jammyjamjamman commented:
> @Jammyjamjamman Do you need any help with the SDL includes?

Thanks Andy. I think the build is working correctly now. Don't think there's much more to do to this PR.

Jammyjamjamman committed:
Tidy up

Jammyjamjamman committed:
More instructions

Jammyjamjamman committed:
Improve ps build script

Whitespaces in dirs doesn't crash it anymore x)
https://stackoverflow.com/a/52140958

Jammyjamjamman committed:
Fix build for latest vcpkg

Jammyjamjamman committed:
Add build instructions

Jammyjamjamman committed:
Add extra notes to readme

Jammyjamjamman committed:
Edit script so it works in gh actions
